source/docs/en/reference.md

```diff
@@ -1,14 +1,52 @@
-# Browser Recommendations
-Please choose a modern browser which is compatible with HTML5,CSS3, and Websocket.  
-Our recommend Browser is Google Chrome.
+# Reference
 
-# Sample Contents List
+## Browser Recommendations
+Please choose a modern browser which is compatible with HTML5, CSS3, and Websockets.  
+We recommend Google Chrome.
 
-#### 1. Algorithm Challenges
-* Languages: C,C++,C#,Java,Javascript,Scala,PHP,Perl,Python,Ruby,Go,Haskell,Groovy
-* contents:FizzBuzz,Primary,etc.
+## Serverside Language and Tool Versions
 
-#### 2. Frameworks Challenges
+When you run tests in our web editor or you submit your answer, we evaluate your code with the below versions of languages and tools.  
+If you're seeing inconsistencies in the test results being run in your local environment and the results displayed on code-check.io, check to see if you are using the exact same dependency versions as our servers.
+
+##### Codecheck CLI  
+- Codecheck v0.5.8
+
+##### Languages
+- NodeJS v4.4.7
+- Ruby v2.3.1p112
+- Python v2.7.11
+- Python3 v3.5.1
+- Java v1.8.0_91
+- Scala v2.11.7
+- Groovy v2.4.5
+- Go v1.6.3
+- PHP v5.6.23-2
+- Haskell v7.6.3
+- perl v5.18.2
+- C/C++ v4.8
+- C# v4.4.1
+
+##### Frameworks
+- Apache Maven v3.0.5
+- sbt v0.13.9
+- Gradle v2.7
+- Composer v1.2.0
+- nosetests v1.3.7
+- mocha v2.5.3
+- PHPUnit v5.4.7
+- Bundler v1.12.5
+- cabal cabal-install v1.22.9.0 using Cabal library v1.22.5.0
+- prove TAP::Harness v3.26 and Perl v5.18.2
+- rspec v3.4.4
+
+## Sample Challenges List
```

Comment:
Since we don't show much of these contents, I'm not sure if we should have these list now.
We should list some challenges that are really published.